third_party/llvm-project/DWARFVisitor.cpp

```diff
@@ -55,6 +55,11 @@ template <typename T> void DWARFYAML::VisitorImpl<T>::traverseDebugInfo() {
       onStartDIE(Unit, Entry);
       if (Entry.AbbrCode == 0u)
         continue;
+      // XXX BINARYEN
+      if (Entry.AbbrCode - FirstAbbrevCode >= DebugInfo.AbbrevDecls.size()) {
+        errs() << "warning: invalid abbreviation code\n";
```

Comment:
Can we add some info to the message e.g. dumping the DIE or something?

Reply:
This is on the YAML code, which doesn't have convenient debug logging, I'm afraid (also in the main dwarf code I don't think you can easily print just one DIE without all the rest, but I'm not sure). But I added logging of the index and the range, which should make it easy to find the value in the dwarfdump output I think.

Reply:
Yeah that looks good.